src/components/articles/AllArticles.jsx

```diff
@@ -0,0 +1,106 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { getAllArticles } from '../../actions';
+
+class AllArticles extends Component {
+  state = {
+    done: false,
+  };
+
+  async componentDidMount() {
+    const { getAllArticles: getAll } = this.props;
+    await getAll();
+    this.setState({ done: true });
+  }
+
+  handleClick = async () => {
+    const { getAllArticles: getAll } = this.props;
+    await getAll();
+    this.setState({ done: true });
+  };
+
+  renderArticles() {
```

Comment:
I saw that you were calling this on line 78, have you thought about making the elements in this method a component on its own? I would like to get your response on this